app/assets/src/components/views/PipelineViz.jsx

```diff
@@ -0,0 +1,379 @@
+import React from "react";
+import { forEach } from "lodash";
+import PropTypes from "prop-types";
+import ReactPanZoom from "@ajainarayanan/react-pan-zoom";
+
+import RemoveIcon from "~/components/ui/icons/RemoveIcon";
+import NetworkGraph from "~/components/visualizations/NetworkGraph.js";
+import cs from "./pipeline_viz.scss";
+
+const START_NODE_ID = -1;
+const END_NODE_ID = -2;
+
+class PipelineViz extends React.Component {
+  constructor(props) {
+    super(props);
+    this.pipelineVersion = this.props.stageResults.pipeline_version;
+    this.stageGraphsInfo = [];
+    this.graphs = [];
+
+    this.stageNames = [
+      "Host Filtering",
+      "GSNAPL/RAPSEARCH alignment",
+      "Post Processing",
+    ];
+    if (this.props.admin) {
+      this.stageNames.push("Experimental");
+    }
+
+    this.state = {
+      stagesOpened: [true, true, true, true],
+      zoom: 1,
+    };
+
+    this.handleMouseWheelZoom = this.handleMouseWheelZoom.bind(this);
+  }
+
+  componentDidMount() {
+    this.drawGraphs();
+  }
+
+  handleMouseWheelZoom(e) {
+    const { zoomChangeInterval } = this.props;
+    const zoomChange = (e.deltaY < 0 ? 1 : -1) * zoomChangeInterval;
+    this.setState({ zoom: this.state.zoom + zoomChange });
+  }
+
+  toggleStage(index) {
+    const updatedStagesOpened = [...this.state.stagesOpened];
+    updatedStagesOpened[index] = !updatedStagesOpened[index];
+    this.setState({ stagesOpened: updatedStagesOpened });
+  }
+
+  modifyStepNames() {
+    // Strips 'PipelineStep[Run/Generate]' from front of each step name.
+    // TODO(ezhong): Consider adding 'name' field to dag_json later.
+    const { stageResults } = this.props;
+    forEach(stageResults.stages, (stageData, _) => {
+      stageData.steps.forEach(step => {
```

Comment:
while I'm here, I figured I could help in the review - You should avoid mutating props in react. If you need to change those, you can create a new array and pass it up through a callback. More likely you just need the modified version from it (don't need to pass it back up).

Reply:
if steps is an array, you can use `stageData.steps.map` otherwise you can import `mapValues` from lodash. Btw, I think it's great you're importing single functions from lodash rather than the whole shebang, much easier to replace if necessary!

Reply:
https://reactjs.org/docs/components-and-props.html#props-are-read-only